### ISSUE TITLE
no-useless-escape false positive for [\w]

### ISSUE THREAD

**Issue by burabure** [eslint/eslint#7769]
**Tell us about your environment**

* **ESLint Version:** 3.9.1
* **Node Version:** 6.9.1
* **npm Version:** 3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

```
parser: babel-eslint
extends: airbnb
```

**What did you do? Please include the actual source code causing the issue.**

```js
RegExp(`${escapeForRegExp(globalExport)}([^\w])`,'g')
```

**What did you expect to happen?**

`[\w]` matches any word character (equal to [a-zA-Z0-9_]), `[w]` matches the character "w", so it shouldn't get flagged as an useless escape

**What actually happened? Please include the actual, raw output from ESLint.**

`[^\w]` got an `no-useless-escape` linting error

**Comment by not-an-aardvark** [comment]
Your code is passing a *string* to the `RegExp` constructor, and `\w` is the same as `w` in strings. So your code will behave the same as

```js
RegExp(`${escapeForRegExp(globalExport)}([^w])`,'g')
```

I assume your code is supposed to look like this instead:

```js
RegExp(`${escapeForRegExp(globalExport)}([^\\w])`,'g')
```

**Comment by burabure** [comment]
whoops, you're right, closing this
